code/modules/projectiles/guns/energy/nuclear.dm

```diff
@@ -54,12 +54,10 @@
 /obj/item/gun/energy/gun/nuclear/on_update_icon()
 	indicator_color = get_charge_color()
 	. = ..()
-	var/list/new_overlays = list()
-	var/reactor_icon = fail_counter ? "danger" : "clean"
-	new_overlays += mutable_appearance(icon, "[get_world_inventory_state()]_[reactor_icon]")
-	var/datum/firemode/current_mode = firemodes[sel_mode]
-	new_overlays += mutable_appearance(icon, "[get_world_inventory_state()]_[current_mode.name]")
-	overlays += new_overlays
+	add_overlay("[get_world_inventory_state()]_[fail_counter ? "danger" : "clean"]")
+	var/datum/firemode/current_mode = LAZYACCESS(firemodes, sel_mode)
+	if(current_mode)
```

Comment:
I believe as discussed in several issues here, the issue is update_icon() for nuclear guns getting called prior to `/obj/item/gun/Initialize()` initializing the firemode.
![image](https://github.com/NebulaSS13/Nebula/assets/110272328/383f0009-0054-4a80-b17a-7c2d0e9d5ea3)
The root cause here is that guns can update their icon prior to firemode init, which should probably be addressed instead of/in addition to fixing it here. (I'd just move the firemode init prior to the parent call.)

Reply:
Fix is in https://github.com/NebulaSS13/Nebula/pull/3209. This commit and the one after it are probably no longer needed, and I don't even think this one would have fixed it anyway since `current_mode` isn't checked to see if it's a firemode datum instead of a list.

Reply:
The crash check commit is still unnecessary, since we know what caused the issue and have a fix for it—I believe I pointed the issue out back in March when my OpenDream stuff found it.